### PR TITLE
Use ansible-vault to encrypt the passwords in tests_sanity_auth.yml and tests_sanity_bpftrace.yml.

### PR DESCRIPTION
**Pull request by nhosoi**
To run `tests_sanity_auth.yml` or `tests_sanity_bpftrace.yml` without
using `tox-lsr`, specify the env var `ANSIBLE_VAULT_PASSWORD_FILE` and
pass `--extra-vars=@path/to/vars/vault-variables.yml` to the command
line as follows.
```
cd tests
ANSIBLE_VAULT_PASSWORD_FILE=vault_pwd TEST_SUBJECTS=/path/to/$image \
      ansible-playbook -vv --extra-vars=@vars/vault-variables.yml \
      -i /path/to/standard-inventory-qcow2 tests_sanity_auth.yml

### CODE REVIEW

**Comment by nhosoi** [comment]
I guess we'd better to wait for the tox-lsr upgraded to the latest including (#90) and (#91).
```
tox / python (3.8) (pull_request) Failing after 5m 
Error:     raise AnsibleVaultError('Attempting to decrypt but no vault secrets found')
```

**Comment by richm** [comment]
```
ansible-plugin-scan run-test: commands[1] | python /home/runner/work/metrics/metrics/.tox/ansible-plugin-scan/report-modules-plugins.py --details .
...
  File "/home/runner/work/metrics/metrics/.tox/ansible-plugin-scan/lib/python3.8/site-packages/ansible/parsing/vault/__init__.py", line 678, in decrypt_and_get_vault_id
Error:     raise AnsibleVaultError('Attempting to decrypt but no vault secrets found')
ansible.parsing.vault.AnsibleVaultError: Attempting to decrypt but no vault secrets found
```
I guess we'll have to make `ANSIBLE_VAULT_PASSWORD_FILE` set in the plugin scan environment . . .

**Comment by richm** [comment]
> ```
> ansible-plugin-scan run-test: commands[1] | python /home/runner/work/metrics/metrics/.tox/ansible-plugin-scan/report-modules-plugins.py --details .
> ...
>   File "/home/runner/work/metrics/metrics/.tox/ansible-plugin-scan/lib/python3.8/site-packages/ansible/parsing/vault/__init__.py", line 678, in decrypt_and_get_vault_id
> Error:     raise AnsibleVaultError('Attempting to decrypt but no vault secrets found')
> ansible.parsing.vault.AnsibleVaultError: Attempting to decrypt but no vault secrets found
> ```
> 
> I guess we'll have to make `ANSIBLE_VAULT_PASSWORD_FILE` set in the plugin scan environment . . .

https://github.com/linux-system-roles/auto-maintenance/pull/187

**Comment by richm** [comment]
[citest]

**Comment by nhosoi** [comment]
[citest bad]

**Comment by nhosoi** [comment]
@natoscott, @richm, tests that enable the grafana server fail on rhel-9.2.

RHEL-9.2.0-20220922.0/ansible-2.13/(citool) — Test finished
[Details](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/lsr-citool_metrics-125-6e54e82_RHEL-9.2.0-20220922.0_20220924-021535/artifacts/summary.html)
@tft-bot
RHEL-9.2.0-20220922.0/ansible-2.9/(citool) — Test finished
[Details](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/lsr-citool_metrics-125-6e54e82_RHEL-9.2.0-20220922.0_20220924-021541/artifacts/summary.html)

Indeed, `grafana-server` fails to start.
```
# systemctl start grafana-server
Job for grafana-server.service failed because the control process exited with error code.
See "systemctl status grafana-server.service" and "journalctl -xeu grafana-server.service" for details.
```
These command lines do not report much :(
```
systemctl status grafana-server.service
journalctl -xeu grafana-server.service
```
But the `/var/log/messages` file stores these messages.
```
Sep 24 20:02:30 localhost grafana-server[27098]: logger=server t=2022-09-24T20:02:30.280933074-04:00 level=info msg="Writing PID file" path=/var/run/grafa
na/grafana-server.pid pid=27098
Sep 24 20:02:30 localhost grafana-server[27098]: logger=provisioning t=2022-09-24T20:02:30.342437066-04:00 level=error msg="Failed to provision plugins" error="app provisioning error: plugin not installed: \"performancecopilot-pcp-app\""
Sep 24 20:02:30 localhost grafana-server[27098]: Failed to start grafana. error: app provisioning error: plugin not installed: "performancecopilot-pcp-app"
Sep 24 20:02:30 localhost grafana-server[27098]: app provisioning error: plugin not installed: "performancecopilot-pcp-app"
Sep 24 20:02:30 localhost systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
Sep 24 20:02:30 localhost systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Sep 24 20:02:30 localhost systemd[1]: Failed to start Grafana instance.
Sep 24 20:02:30 localhost systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 1.
```
Is this a known issue for rhel-9.2? Could you please advise?

**Comment by natoscott** [comment]
@andreasgerstmayr do you recognise this one Andreas?
@nhosoi I believe that fresh 9.2 rebases for grafana and grafana-pcp packages have been built in the last few days, so this may be related to those updates.

**Comment by nhosoi** [comment]
[citest bad]

**Comment by nhosoi** [comment]
> tl;dr can you re-test with the latest compose?

Thanks, @andreasgerstmayr.

The current latest build is `RHEL-9.2.0-20220923.0` and I guess it does not contain the fix? It'd be nice if we could learn the build date which was fixed...

**Comment by andreasgerstmayr** [comment]
@nhosoi this is the build: https://kojihub.stream.rdu2.redhat.com/koji/buildinfo?buildID=25054
Completed on Tue, 20 Sep 2022 15:59:59 UTC.

**Comment by andreasgerstmayr** [comment]
@nhosoi: The grafana-pcp build is currently being tested by QE. Once that process is finished, it will land in the next compose.

**Comment by richm** [comment]
[citest]

**Comment by nhosoi** [comment]
Hi @andreasgerstmayr,

> @nhosoi: the latest compose RHEL-9.2.0-20221003.0 contains grafana-pcp v5 - let me know if there are still issues.

Unfortunately, the tests are still failing in my local testing.

My 9.2 image is `rhel-guest-image-9.2-20221003.0.x86_64.qcow2` from the download site.

`grafana-server` fails with `plugin not installed: "performancecopilot-pcp-app"`.
```
Oct  3 14:17:22 localhost grafana-server[25906]: logger=server t=2022-10-03T14:17:22.869258643-04:00 level=info msg="Writing PID file" path=/var/run/grafana/grafana-server.pid pid=25906
Oct  3 14:17:22 localhost grafana-server[25906]: logger=provisioning t=2022-10-03T14:17:22.925894711-04:00 level=error msg="Failed to provision plugins" error="app provisioning error: plugin not installed: \"performancecopilot-pcp-app\""
Oct  3 14:17:22 localhost grafana-server[25906]: Failed to start grafana. error: app provisioning error: plugin not installed: "performancecopilot-pcp-app"
Oct  3 14:17:22 localhost grafana-server[25906]: app provisioning error: plugin not installed: "performancecopilot-pcp-app"
Oct  3 14:17:22 localhost systemd[1]: grafana-server.service: Main process exited, code=exited, status=1/FAILURE
```
`grafana` related versions:
```
# rpm -qa | egrep grafana
grafana-pcp-5.0.0-4.el9.x86_64
grafana-9.0.9-1.el9.x86_64
```
Obviously, `performancecopilot-pcp-app` is indeed installed on the managed host.
```
# ls /var/lib/grafana/plugins/performancecopilot-pcp-app/
CHANGELOG.md  fonts                  monaco-editor.js             plugin.json
LICENSE       img                    monaco-editor.js.map         vendors~monaco-editor.js
README.md     module.js              monaco-editor.worker.js      vendors~monaco-editor.js.LICENSE.txt
dashboards    module.js.LICENSE.txt  monaco-editor.worker.js.map  vendors~monaco-editor.js.map
datasources   module.js.map          panels
```
I played some more with sysconfig. The homedir of `grafana-server` is here:
```
GRAFANA_HOME=/usr/share/grafana
```
in which directory, there is `plugins-bundled`. I created a symlink like this for fun:
```
# ls -l /usr/share/grafana/plugins-bundled/internal/performancecopilot-pcp-app
lrwxrwxrwx. 1 root root 52 Oct  3 14:35 /usr/share/grafana/plugins-bundled/internal/performancecopilot-pcp-app -> /var/lib/grafana/plugins/performancecopilot-pcp-app/
```
Then, `grafana-server` successfully started.
```
# systemctl status grafana-server
● grafana-server.service - Grafana instance
     Loaded: loaded (/usr/lib/systemd/system/grafana-server.service; enabled; vendor preset: disabled)
     Active: active (running) since Mon 2022-10-03 14:36:23 EDT; 4min 47s ago
        <<snip>>
```
I don't know this is the right configuration or not. But at least, it looks to me `grafana-server` still fails to access `performancecopilot-pcp-app`... I'm wondering `rhel-guest-image-9.2-20221003.0.x86_64.qcow2` does not contain the fix you mentioned? Or could there be some more issues?

**Comment by nhosoi** [comment]
[citest bad]

**Comment by andreasgerstmayr** [comment]
I figured the root cause - Grafana's plugin signature validation got more strict (needs all plugin components in the allowed list vs just the 'root' plugin), and while this change is contained in Fedora/CentOS/RHEL Grafana rpm's default configuration (therefore it works fine when installing grafana and grafana-pcp on a RHEL 9.2 compose), it gets overwritten with a different `grafana.ini` config file when using this ansible role.

I've created https://github.com/performancecopilot/ansible-pcp/pull/39 to fix this.

> in which directory, there is `plugins-bundled`. I created a symlink like this for fun:
> 
> ```
> # ls -l /usr/share/grafana/plugins-bundled/internal/performancecopilot-pcp-app
> lrwxrwxrwx. 1 root root 52 Oct  3 14:35 /usr/share/grafana/plugins-bundled/internal/performancecopilot-pcp-app -> /var/lib/grafana/plugins/performancecopilot-pcp-app/
> ```
> 
> Then, `grafana-server` successfully started.

That's interesting, looks like for this folder the signature validation gets ignored.
Basically:

* `/usr/share/grafana/plugins-bundled/internal` is a folder for plugins bundled by Grafana itself
* `/var/lib/grafana/plugins` is the folder where Grafana installs plugins (either by the UI or by `grafana-cli`)
* `/usr/share/performancecopilot-pcp-app` is the folder where the grafana-pcp RPM is getting installed, and systemd-tmpfiles creates a symlink from `/var/lib/grafana/plugins` to this folder. This symlink magic is required because on rpm-ostree systems (Fedora Silverblue and RHEL Edge for example), packages are not allowed to package files in `/var` (`/var` is reserved exclusively for application state).


> I'm wondering `rhel-guest-image-9.2-20221003.0.x86_64.qcow2` does not contain the fix you mentioned? Or could there be some more issues?

Sorry for the confusion - initially I assumed it's just an incompatibility of Grafana v9 (got upgraded from v7) and grafana-pcp v3, but the root cause (and fix) is described above.

**Comment by nhosoi** [comment]
Thank you soooooo much, @andreasgerstmayr !
I applied your https://github.com/performancecopilot/ansible-pcp/pull/39 and ran the tests in my test env. The tests passed beautifully.

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
please rebase

**Comment by nhosoi** [comment]
> please rebase

Done!

**Comment by nhosoi** [comment]
[citest]
